### ISSUE TITLE
Completing enum variants fills with parentheses

### ISSUE THREAD

**Issue by ntBre** [rust-lang/rust-analyzer#13767]
Currently, enum variants get completed with parentheses inside the variant. This completion also leaves the cursor after the closing parentheses, forcing me to move back into the parentheses, delete everything inside, and then type what I actually want in there.

![Example of the issue](https://user-images.githubusercontent.com/36778786/207358102-36a06c1f-246b-41b8-b6c1-ffd1ace2a935.gif)

I would much rather the completion stop at `Enum::Variant`, allowing me to type the parentheses and their contents, or at most, for the completion snippet to leave me inside the parentheses. Could some kind of configuration option be added for this? If someone could point me in the right direction, I would be glad to try to work on this myself too.

I initially posted this as an Emacs [lsp-mode issue](https://github.com/emacs-lsp/lsp-mode/issues/3852) but was pointed over here.

**Comment by flodiebold** [comment]
IMO if snippets aren't enabled we shouldn't be adding any parentheses at all, as we do for function calls; if snippets _are_ enabled, we should still have a setting like for function calls to control whether to add snippets, parentheses or nothing. Or maybe we should just respect the callable snippet setting.

For functions this happens here:
https://github.com/rust-lang/rust-analyzer/blob/4596847a88abb0d5077c5111c3093e724673d7a0/crates/ide-completion/src/render/function.rs#L153
note that this function already only gets called if snippets are enabled.
The code that would need to be changed is here:
https://github.com/rust-lang/rust-analyzer/blob/4596847a88abb0d5077c5111c3093e724673d7a0/crates/ide-completion/src/render/variant.rs#L45-L65

**Comment by ntBre** [comment]
> IMO if snippets aren't enabled we shouldn't be adding any parentheses at all, as we do for function calls; if snippets _are_ enabled, we should still have a setting like for function calls to control whether to add snippets, parentheses or nothing. Or maybe we should just respect the callable snippet setting.

I definitely agree. I have `lsp-rust-analyzer-completion-add-call-parenthesis` set to `nil`, and I think I have snippets enabled otherwise, so it would be great to have a similar setting for this.

Thank you very much for the code pointers! I'll try to dive in later today.

**Comment by ntBre** [comment]
Sorry for the delay on this, I finally got some time to look into it. I added a commit that handles the very basic version of what I wanted, but it certainly doesn't cover everything we've talked about. I also couldn't figure out how to update the label on the completion, so it still says "Variant(...)" even though it only completes "Variant" as I wanted. 

I guess I will need to make the signature of `render_tuple_lit` more like that of `add_call_parens` and pass more information around. Does that sound right to you? Thanks again for all of the pointers!
